### ISSUE TITLE
gRPC tag_values asserts: assertion failed: !partition_predicate.has_field_restriction()'

### ISSUE THREAD

**Issue by alamb** [influxdata/influxdb_iox#518]
__Steps to reproduce:__
Select multiple _field restrictions in the UI such as in this screenshot:

![Screen Shot 2020-12-02 at 8 33 38 AM](https://user-images.githubusercontent.com/490673/100879177-23b9d300-3479-11eb-9b00-132b7c4c7600.png)

__Expected behaviour:__
A response to be correctly generated

__Actual behaviour:__

There is a panic in the log
```
[2020-12-02T13:32:06Z INFO  influxdb_iox::server::rpc::storage] tag_values for database 26f7e5a4b7be365b_917b97a92e883afc, range: Some(TimestampRange { start: 1606915026289397000, end: 1606915926289397000 }), tag_key: undefined, predicate: (((TagRef:_m[0x00] == "cpu") AND ((TagRef:_f[0xff] == "usage_user") Or (TagRef:_f[0xff] == "usage_system"))) AND (TagRef:cpu == "cpu-total")) 
[2020-12-02T13:32:06Z ERROR influxdb_iox::panic] thread 'tokio-runtime-worker' panicked at 'assertion failed: !partition_predicate.has_field_restriction()', write_buffer/src/table.rs:419:9 
thread 'tokio-runtime-worker' panicked at 'assertion failed: !partition_predicate.has_field_restriction()', write_buffer/src/table.rs:419:9
```

__Config:__

I am running IOx in this way:
```
cd /Users/alamb/Software/influxdb_iox && INFLUXDB_IOX_BIND_ADDR=127.0.0.1:11080 cargo run -- -v
```

**Comment by alamb** [comment]
I did a little more investigation. I don't have any useful respond


The gRPC query looks like this (indents are mine):

[2020-12-02T13:32:06Z INFO  influxdb_iox::server::rpc::storage] tag_values for database 26f7e5a4b7be365b_917b97a92e883afc, range: Some(TimestampRange { start: 1606915026289397000, end: 1606915926289397000 }), tag_key: undefined, predicate: (((TagRef:_m[0x00] == "cpu") AND ((TagRef:_f[0xff] == "usage_user") Or (TagRef:_f[0xff] == "usage_system"))) AND (TagRef:cpu == "cpu-total"))

```
tag_values for database 26f7e5a4b7be365b_917b97a92e883afc,
  range: Some(TimestampRange { start: 1606915026289397000, end: 1606915926289397000 }),
  tag_key: undefined,
  predicate:
  (
    (
      (TagRef:_m[0x00] == "cpu")
      AND
      (
        (TagRef:_f[0xff] == "usage_user")
        Or (TagRef:_f[0xff] == "usage_system")
      )
    )
    AND
    (TagRef:cpu == "cpu-total"))
```



When I select just two fields, then the query looks like this (which is fine)

[2020-12-02T19:53:11Z INFO  influxdb_iox::server::rpc::storage] tag_values for database 26f7e5a4b7be365b_917b97a92e883afc, range: Some(TimestampRange { start: 1606937891192224000, end: 1606938791192224000 }), tag_key: cpu, predicate: ((TagRef:_m[0x00] == "cpu") AND ((TagRef:_f[0xff] == "usage_user") Or (TagRef:_f[0xff] == "usage_system")))

When then I further select a a cpu=cpu-total tag that is when we see the predicate

There are actual two request that get made that panic (both tag_keys and tag_values):

```
[2020-12-02T19:53:56Z INFO  influxdb_iox::server::rpc::storage] tag_keys for database 26f7e5a4b7be365b_917b97a92e883afc, range: Some(TimestampRange { start: 1606937936606532000, end: 1606938836606532000 }), predicate: (((TagRef:_m[0x00] == "cpu") AND ((TagRef:_f[0xff] == "usage_user") Or (TagRef:_f[0xff] == "usage_system"))) AND (TagRef:cpu == "cpu-total"))
[2020-12-02T19:53:56Z ERROR influxdb_iox::panic] thread 'tokio-runtime-worker' panicked at 'assertion failed: !partition_predicate.has_field_restriction()', write_buffer/src/table.rs:349:9
thread 'tokio-runtime-worker' panicked at 'assertion failed: !partition_predicate.has_field_restriction()', write_buffer/src/table.rs:349:9
```

```
[2020-12-02T19:53:56Z INFO  influxdb_iox::server::rpc::storage] tag_values for database 26f7e5a4b7be365b_917b97a92e883afc, range: Some(TimestampRange { start: 1606937936677017000, end: 1606938836677017000 }), tag_key: undefined, predicate: (((TagRef:_m[0x00] == "cpu") AND ((TagRef:_f[0xff] == "usage_user") Or (TagRef:_f[0xff] == "usage_system"))) AND (TagRef:cpu == "cpu-total"))
[2020-12-02T19:53:56Z ERROR influxdb_iox::panic] thread 'tokio-runtime-worker' panicked at 'assertion failed: !partition_predicate.has_field_restriction()', write_buffer/src/table.rs:419:9
thread 'tokio-runtime-worker' panicked at 'assertion failed: !partition_predicate.has_field_restriction()', write_buffer/src/table.rs:419:9
[2020-12-02T19:54:01Z INFO  influxdb_iox::server::http_routes] Handled request method=POST uri=/api/v2/write?bucket=917b97a92e883afc&org=26f7e5a4b7be365b status=204
```

**Comment by alamb** [comment]
Here is what is returned from the existing storage engine when I choose these restrictions in the UI: ` _measurement=cpu, _field=usage_system+usage_user`:

<img width="1730" alt="Screen Shot 2020-12-02 at 4 07 19 PM" src="https://user-images.githubusercontent.com/490673/100931911-2805e080-34b9-11eb-927d-5a924319e902.png">

```
ts=2020-12-02T21:01:02.832251Z lvl=info msg="AAL TagKeys" log_id=0QqJw7ZW000 component=read
ts=2020-12-02T21:01:02.832285Z lvl=info msg="AAL   TagKeys source partition_id: 0 org_id: 26f7e5a4b7be365b bucket_id: 917b97a92e883afc" log_id=0QqJw7ZW000 component=read
ts=2020-12-02T21:01:02.832290Z lvl=info msg="AAL   TagKeys pred '\u0000' = \"cpu\" AND '\ufffd' = \"usage_guest\" OR '\ufffd' = \"usage_user\" OR '\ufffd' = \"usage_system\"" log_id=0QqJw7ZW000 component=read
ts=2020-12-02T21:01:02.832298Z lvl=info msg="AAL   TagKeys range {1606941962817848000 1606942862817848000}" log_id=0QqJw7ZW000 component=read
AAL   WriteStringIterator 
AAL   WriteStringIterator cpu
AAL   WriteStringIterator host
AAL   WriteStringIterator ˇ
```

and

```
ts=2020-12-02T21:01:02.921238Z lvl=info msg="AAL TagValues" log_id=0QqJw7ZW000 component=read
ts=2020-12-02T21:01:02.921269Z lvl=info msg="AAL   TagValues source partition_id: 0 org_id: 26f7e5a4b7be365b bucket_id: 917b97a92e883afc" log_id=0QqJw7ZW000 component=read
ts=2020-12-02T21:01:02.921274Z lvl=info msg="AAL   TagValues pred '\u0000' = \"cpu\" AND '\ufffd' = \"usage_guest\" OR '\ufffd' = \"usage_user\" OR '\ufffd' = \"usage_system\"" log_id=0QqJw7ZW000 component=read
ts=2020-12-02T21:01:02.921281Z lvl=info msg="AAL   TagValues range {1606941962910738000 1606942862910738000}" log_id=0QqJw7ZW000 component=read
ts=2020-12-02T21:01:02.921284Z lvl=info msg="AAL   TagValues tag_key cpu" log_id=0QqJw7ZW000 component=read
AAL   WriteStringIterator cpu-total
AAL   WriteStringIterator cpu0
AAL   WriteStringIterator cpu1
AAL   WriteStringIterator cpu10
AAL   WriteStringIterator cpu11
AAL   WriteStringIterator cpu12
AAL   WriteStringIterator cpu13
AAL   WriteStringIterator cpu14
AAL   WriteStringIterator cpu15
AAL   WriteStringIterator cpu2
AAL   WriteStringIterator cpu3
AAL   WriteStringIterator cpu4
AAL   WriteStringIterator cpu5
AAL   WriteStringIterator cpu6
AAL   WriteStringIterator cpu7
AAL   WriteStringIterator cpu8
AAL   WriteStringIterator cpu9
```


Now, when I also add the `cpu=cpu-total` (aka put a real tag restriction on it), for a total of` _measurement=cpu, _field=usage_system+usage_user, cpu=cpu-total`:

<img width="1792" alt="Screen Shot 2020-12-02 at 4 09 04 PM" src="https://user-images.githubusercontent.com/490673/100931969-3653fc80-34b9-11eb-8192-6b8c362a80bb.png">

Here is what comes back

```
ts=2020-12-02T21:02:20.049042Z lvl=info msg="AAL TagKeys" log_id=0QqJw7ZW000 component=read
ts=2020-12-02T21:02:20.049074Z lvl=info msg="AAL   TagKeys source partition_id: 0 org_id: 26f7e5a4b7be365b bucket_id: 917b97a92e883afc" log_id=0QqJw7ZW000 component=read
ts=2020-12-02T21:02:20.049078Z lvl=info msg="AAL   TagKeys pred '\u0000' = \"cpu\" AND '\ufffd' = \"usage_guest\" OR '\ufffd' = \"usage_user\" OR '\ufffd' = \"usage_system\" AND 'cpu' = \"cpu-total\"" log_id=0QqJw7ZW000 component=read
ts=2020-12-02T21:02:20.049097Z lvl=info msg="AAL   TagKeys range {1606942040035331000 1606942940035331000}" log_id=0QqJw7ZW000 component=read
AAL   WriteStringIterator 
AAL   WriteStringIterator cpu
AAL   WriteStringIterator host
AAL   WriteStringIterator ˇ
```


```
ts=2020-12-02T21:02:20.142090Z lvl=info msg="AAL TagValues" log_id=0QqJw7ZW000 component=read
ts=2020-12-02T21:02:20.142122Z lvl=info msg="AAL   TagValues source partition_id: 0 org_id: 26f7e5a4b7be365b bucket_id: 917b97a92e883afc" log_id=0QqJw7ZW000 component=read
ts=2020-12-02T21:02:20.142127Z lvl=info msg="AAL   TagValues pred '\u0000' = \"cpu\" AND '\ufffd' = \"usage_guest\" OR '\ufffd' = \"usage_user\" OR '\ufffd' = \"usage_system\" AND 'cpu' = \"cpu-total\"" log_id=0QqJw7ZW000 component=read
ts=2020-12-02T21:02:20.142135Z lvl=info msg="AAL   TagValues range {1606942040130039000 1606942940130039000}" log_id=0QqJw7ZW000 component=read
ts=2020-12-02T21:02:20.142141Z lvl=info msg="AAL   TagValues tag_key host" log_id=0QqJw7ZW000 component=read
AAL   WriteStringIterator MacBook-Pro.hsd1.ma.comcast.net
```

** in other words, this looks like tag_keys returns tags for tables that have those field present.
